internal/services/recoveryservices/site_recovery_replicated_vm_resource.go

```diff
@@ -239,6 +258,13 @@ func resourceSiteRecoveryReplicatedItemCreate(d *pluginsdk.ResourceData, meta in
 		targetAvailabilitySetID = nil
 	}
 
+	var targetAvailabilityZone *string
+	if id, isSet := d.GetOk("target_availability_zone"); isSet {
+		targetAvailabilityZone = utils.String(id.(string))
```

Comment:
Maybe change `id` to `zone`?
```suggestion
	if zone, isSet := d.GetOk("target_availability_zone"); isSet {
		targetAvailabilityZone = utils.String(zone.(string))
```

Reply:
That makes more sense, thanks for the review!